src/lib/util/constants.js

```diff
@@ -45,38 +45,112 @@ exports.DEFAULTS = {
 		usage: ''
 	},
 
-	GATEWAY_GUILDS_RESOLVER: async function validateGuild(guildResolvable) {
-		if (guildResolvable) {
-			let value;
+	GATEWAY_GUILDS: {
+		name: 'guilds',
+		resolver: async function validateGuild(guildResolvable) {
+			if (guildResolvable) {
+				let value;
 
-			if (typeof guildResolvable === 'string' && /^\d{17,19}$/.test(guildResolvable)) value = this.client.guilds.get(guildResolvable);
-			else if (guildResolvable instanceof Guild) value = guildResolvable;
-			if (value) return value;
-		}
+				if (typeof guildResolvable === 'string' && /^\d{17,19}$/.test(guildResolvable)) value = this.client.guilds.get(guildResolvable);
+				else if (guildResolvable instanceof Guild) value = guildResolvable;
+				if (value) return value;
+			}
 
-		throw new Error('The parameter <Guild> expects either a Guild ID or a Guild Instance.');
+			throw new Error('The parameter <Guild> expects either a Guild ID or a Guild Instance.');
+		},
+		schema: {
+			prefix: {
+				type: 'string',
+				default: this.client.options.prefix,
```

Comment:
The `this` keyword has no context here, therefore doubtfully it'll work. That's why I kept them in GatewayDriver. Resolvers are fine because of this line:

https://github.com/dirigeants/klasa/blob/531b938b0388a5719510428730d321da059ad17b/src/lib/settings/GatewayDriver.js#L160